private/buf/bufwire/module_config_reader.go

```diff
@@ -622,6 +622,7 @@ func (m *moduleConfigReader) getSourceModuleConfig(
 		}
 		buildOptions = append(buildOptions, bufmodulebuild.WithExcludePaths(bucketRelPaths))
 	}
+	buildOptions = append(buildOptions, bufmodulebuild.WithID("dir:"+subDirPath))
```

Comment:
For the IDs, I think we want to expose it as a part of the `Ref`, which means we'll need to expose it from the depths of `buffetch/internal` upwards.
This way, we actually also expose the type of ref this is, since this is not necessarily a `dir` ref.